mangoPURE/data/transforms.py

```diff
@@ -0,0 +1,280 @@
+from datasets import Dataset as HfDataset
+from .base import MixedExample, SegmentInfo, MixerTransform
+import random
+import torch
+from .utils import expand_tensor, calc_audio_adjustment_coef
+
+
+class CreateRandomBlankAudio(MixerTransform):
+    def __init__(
+            self,
+            min_secs: float = 10.0,
+            max_secs: float = 20.0,
+            rate: int = 16000
+    ):
+        self.min_len = int(rate * min_secs)
+        self.max_len = int(rate * max_secs)
+
+    def __call__(
+            self,
+            example: MixedExample,
+    ) -> MixedExample:
+        """
+        It creates MixedExample with random one speaker audio
+        """
+        random_len = random.randint(self.min_len, self.max_len)
+        audio = torch.zeros(random_len)
+        mixed_example = MixedExample(
+            audio=audio
+        )
+        return mixed_example
+
+
+class CreateRandomSpeaker(MixerTransform):
+    def __init__(self, speaker_dataset: HfDataset):
+        self.speaker_dataset = speaker_dataset
+
+    def __call__(
+            self,
+            example: MixedExample,
+    ) -> MixedExample:
+        """
+        It creates MixedExample with random one speaker audio
+        """
+        random_pos = random.randint(0, len(self.speaker_dataset) - 1)
+        speaker_example = self.speaker_dataset[random_pos]
+        audio = speaker_example["audio"]["array"]
+
+        mixed_example = MixedExample(
+            audio=audio,
+            speakers_info=[SegmentInfo(
+                start=0,
+                length=audio.shape[0],
+                class_id=speaker_example["speaker_id"]
```

Comment:
That's not good. It would be better to pass something like `AudioProvider`  that implements 3 methods -
```
AudioProvider:
    len: int
    audio: 1D torch.tensor
    speaker_id: Any 
```
audio and speaker id, ant not assume how this things are named in dataset. This code works for only hf-datasets and only for cv13. If dataset has something named `speaker`  we need to rename column, if dataset have audio as not converted array, we need to convert it. And of corse, if dataset is not hf dataset everything does not work. 

Further, we can store it in some `providers` packages and chose the one we need depending on class of object passed